hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -1601,7 +1601,7 @@ public void rename(final Path source, final Path destination,
   }
 
   private Boolean isCreateOperationOnBlobEndpoint() {
-    return !OperativeEndpoint.isIngressEnabledOnDFS(prefixMode, abfsConfiguration);
+    return !OperativeEndpoint.isIngressEnabledOnDFS(getPrefixMode(), abfsConfiguration);
```

Comment:
I was not able to follow why we need to send prefix mode as a separate argument here??
As per offline discussion you mentioned that 

"The intent on sending the prefixMode as separate param was for the fallback mechanism we implemeted in append and flush flow. If a request we fallback to DFS we don't want to change the global prefix mode only at abfsOutputStream level. Hence I had added prefixMode as a separate param"

Is there a scenario where we are changing the prefix mode locally for a call??

Reply:
yes when append over blob fails with a 409 blob operation not supported, that means create was done over dfs endpoint and hence append and flush also need to go over dfs. So in AbfsOuputStream class we change the local variable to have prefix mode as dfs.

Reply:
Makes sense...
Thanks